nemo/collections/nlp/models/lm_model.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Dict, List, Optional
+
+import torch
+import torch.nn as nn
+from torch.utils.data import DataLoader
+
+from nemo.collections.common.losses import AggregatorLoss, CrossEntropyLoss
+from nemo.collections.common.tokenizers.bert_tokenizer import NemoBertTokenizer
+from nemo.collections.nlp.data.lm_bert_dataset import BertPretrainingPreprocessedDataloader
+from nemo.collections.nlp.losses import SmoothedCrossEntropyLoss
+from nemo.collections.nlp.modules.common import SequenceClassifier, TokenClassifier
+from nemo.collections.nlp.modules.common.common_utils import get_pretrained_lm_model
+from nemo.collections.nlp.modules.common.huggingface.bert import BertEncoder
+from nemo.core.classes import typecheck
+from nemo.core.classes.modelPT import ModelPT
+from nemo.core.neural_types import NeuralType
+from nemo.utils.decorators import experimental
+
+__all__ = ['BERTLMModel']
+
+
+@experimental
+class BERTLMModel(ModelPT):
+    """
+    BERT LM model pretraining.
+    """
+
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.bert_model.input_types
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return {
+            'mlm_logits': self.mlm_classifier.output_types['logits'],
+            'nsp_logits': self.nsp_classifier.output_types['logits'],
+        }
+
+    def __init__(self, config_file: Optional[str] = None, pretrained_model_name: Optional[str] = 'bert-base-uncased'):
```

Comment:
let's leave it for now, but just a heads up that we won't have config file paths in Model's constructor